lib/bsdlib/nrf91_sockets.c

```diff
@@ -650,6 +653,8 @@ static int nrf91_socket_offload_setsockopt(void *obj, int level, int optname,
 		nrf_rcvtimeo.tv_usec = ((struct timeval *)optval)->tv_usec;
 		nrf_optval = &nrf_rcvtimeo;
 		nrf_optlen = sizeof(struct nrf_timeval);
+	} else if ((level == SOL_TLS) && (optname == TLS_SESSION_CACHE)) {
+		nrf_optlen = sizeof(nrf_sec_session_cache_t);
```

Comment:
so is `optlen` ignored?

Reply:
Yeah basically, handled the same way as the above case.
It will fail if `nrf_optlen` is set to anything other than `sizeof(nrf_sec_session_cache_t)`, but I could add a check for non-zero `optlen`, perhaps?